### ISSUE TITLE
Emit error when overloaded function has an implementation

### ISSUE THREAD

**Issue by hmc-cs-mdrissi** [microsoft/pyright#2074]
**Describe the bug**
```python
from typing import Union, overload


class Foo:
    @overload
    def bar(self, x: int) -> int:
        ...

    @overload
    def bar(self, x: str) -> str:
        ...

    @overload
    def bar(self, x: Union[int, str]) -> Union[int, str]:
        return x

```

**To Reproduce**
Using pylance 2021.7.3-pre.1 I get an error message of,

`"bar" is marked as overload, but no implementation is provided`

for the first bar. The second bar somehow doesn't complain, but the first fails to detect the implementation.

**Expected behavior**
No error.

**VS Code extension or command-line**
pylance 2021.7.3-pre.1

**Comment by JelleZijlstra** [comment]
You should not put the `@overload` decorator on the implementation. Perhaps pyright could give an error message that makes that more clear, but I'm not sure what a better place to put the error would be.

Also, in your example you don't need overloads, since you can use a constrained TypeVar instead. Perhaps your real code is more complicated, though.

**Comment by jakebailey** [comment]
I'm surprised that we don't flag overloaded functions with bodies, but perhaps that's only enabled in the stricter mode.

**Comment by erictraut** [comment]
We do emit an error for overloaded functions without an implementation. We don't currently emit an error for an overloaded function with a body. That's a good suggestion though. I'll add that.

**Comment by hmc-cs-mdrissi** [comment]
Yeah dropping the last overload fixes it. Just got confused by the error message.

**Comment by erictraut** [comment]
The new check will be implemented in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.156, which I just published. It will also be included in the next release of pylance.
